### PR TITLE
Refactor JobPaginator Component

### PR DESCRIPTION
**Pull request by diboy2**
Anything with AjaxHelper I'm probably going to refactor and probably split into a client model and shared state model.

### CODE REVIEW

File: client/src/JobPaginator.jsx [hunk 1]
```diff
@@ -1,326 +1,64 @@
...
+            deletePage,
+            save,
+            autonumberFollowingPages,
+            getImageUrl,
+            setZoom,
+        },
+    } = useJobPaginator(initialCategory, initialJob);
+
+    return (
+        <div className={active ? "" : "hidden"} id="paginator">
```
Comment (demiankatz) [inline client/src/JobPaginator.jsx]: I suspect that "active" can be totally eliminated here, since (as with one of the other components you worked on) it was only here to hide the component in the days before we introduced routing to the application.

File: client/src/JobPaginatorState.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            message.push(added.length + " file(s) have been added to the job since the last edit.\n");
+            setOrder(getAddedPages(order, added));
+        }
+        return message.length ? alert(message.join()) : undefined;
+    };
+
+    const loadJob = async () => {
+        const { order } = await getJob(category, job);
+        setOrder(order);
+        setCategory(category);
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js]: Is there any point to the setCategory and setJob calls here? It seems to me they're just setting the values to their current values, but maybe I'm missing a subtlety.

File: client/src/JobPaginator.jsx (hunk 1 above)
Comment (demiankatz) [reply client/src/JobPaginator.jsx]: (I do see that this is apparently used to hide everything while data is loading, so maybe that's useful... but I'd be interested to see if there's a negative consequence to disabling those calls).

File: client/src/JobPaginatorState.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        setPage(currentPage - 1);
+    };
+
+    const updatePagesByStatus = (status) => {
+        const {
+            file_problems: { deleted, added },
+        } = status;
+        const message = [];
+        if (deleted.length > 0) {
+            message.push(deleted.length + " file(s) have been removed = require(the job since the last edit.\n)");
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js]: This should be "have been removed from the job" -- looks like perhaps a search-and-replace or other IDE feature malfunctioned in the past. (I realize that this problem was not introduced by this PR, but we might as well fix it while we see it). Here's the original code from the previous iteration of the project: https://github.com/FalveyLibraryTechnology/VuDLPrep/blob/f59a7529aa4bb6afecd4508ca5284637a23ee3c9/app/assets/javascripts/components/JobPaginator.js.jsx#L120

File: client/src/JobPaginatorZoom.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import PaginatorPreview from "./PaginatorPreview";
+import PaginatorZoomy from "./PaginatorZoomy";
+
+const JobPaginatorZoom = ({ zoom, getImageUrl }) => {
```
Comment (demiankatz) [inline client/src/JobPaginatorZoom.jsx]: This might need a better name, since it's only a Zoom when zoom is set to true. Maybe JobPaginatorZoomToggle?

Comment (demiankatz) [reply client/src/JobPaginatorZoom.jsx]: Or JobPaginatorTogglableZoom?

File: client/src/JobPaginatorZoom.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    return (
+        <div className="six col">
```
Comment (demiankatz) [inline client/src/JobPaginatorZoom.jsx]: I also wonder whether it's better to factor this outermost `<div>` back into the Job Paginator itself... since this is part of the paginator's layout, it seems like it might be cleaner to keep the layout columns together in the same component so it's easier to adjust them, rather than embedding them in this component (which reduces its ability to be used in a different context).

**Comment by demiankatz** [comment]
@diboy2, I left a few comments, though I still haven't done a fully comprehensive review. I also tried to run the code currently in the branch, but it does not appear to be in a working state quite yet -- I think there are some problems related to the label code. No labels display on any of the thumbnails, and if I try to set a label, I get a fatal error:

<pre>
TypeError: order[imageNumber] is undefined
setLabel
src/JobPaginatorState.js:141
</pre>

I realize this may still be work in progress, so I won't put more time into testing until you give me the go-ahead. In the meantime, I'll leave a couple more comments related to labels which might be helpful.

File: client/src/JobPaginatorState.js [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+        }
+        order[imageNumber]["label"] = text !== null && text.length === 0 ? null : text;
+        setOrder(order);
+        dispatchEvent(new Event("Prep.editted"));
+    };
+
+    const getMagicLabel = (imageNumber) => {
+        const label = getLabel(order, imageNumber);
+
+        if (!label) {
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js]: Is this logic backwards? It seems like it will cause this to always return null (or at least return null under inappropriate circumstances).

I appreciate your efforts to make the getLabel() method less confusing and more explicit, but I think some subtlety has gotten lost along the way. There are really three scenarios that need to be addressed:

1.) We want to know whether or not the user has set a label explicitly on an image.
2.) We want to get the auto-generated "magic label" for a set position.
3.) We want to get the user-generated label if available, or the "magic label" otherwise.

I think in the current code, cases 2 and 3 are getting mixed up and it's causing problems. Maybe we want three functions: getUserLabel, getMagicLabel, and getBestLabel... that might make this all more clear.

File: client/src/Thumbnail.jsx [hunk 7]
```diff
@@ -1,22 +1,18 @@
...
     }, [selected]);
 
     return (
-        <div onClick={selectPage} className={"thumbnail" + (selected ? " selected" : "")} ref={ref}>
-            <img alt="" src={paginator.getImageUrl(number, "thumb")} />
+        <div onClick={() => setPage(number)} className={"thumbnail" + (selected ? " selected" : "")} ref={ref}>
+            <img alt="" src={getImageUrl(number, "thumb")} />
             <div className="number">{number + 1}</div>
-            <div className={"label" + (null === paginator.getLabel(number, false) ? " magic" : "")}>{label}</div>
+            <div className={"label" + (!getMagicLabel(number) ? " magic" : "")}>{getMagicLabel(number)}</div>
```
Comment (demiankatz) [inline client/src/Thumbnail.jsx]: Here's an example of a possible problem with label use. I think what we might really want here is something like:

`<div className={"label" + (null === getUserLabel(number) ? " magic" : "")}>{getBestLabel(number)}</div>`

In other words, display the most appropriate label for the thumbnail, but if it's an auto-generated magic label, apply an extra class so we can style it differently.

File: client/src/JobPaginatorState.js (hunk 3 above)
Comment (demiankatz) [reply client/src/JobPaginatorState.js]: I also realized that the "\n" is in the wrong place in this string (something @crhallberg fixed in #71, but which hasn't been merged anywhere else).

File: client/src/useJobPaginator.js [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    const getMagicLabel = (imageNumber) => {
+        const label = getLabel(order, imageNumber);
+
+        if (label) {
+            return label;
+        }
+        if (typeof magicLabelCache[imageNumber] === "undefined") {
+            magicLabelCache[imageNumber] = MagicLabeler.getLabel(imageNumber, getMagicLabel);
+        }
+        setMagicLabelCache(magicLabelCache);
```
Comment (demiankatz) [inline client/src/useJobPaginator.js]: Is there a reason this was moved out of the if statement above? I would think that we only need to set this value when it changes, but maybe there's a subtlety I'm not aware of.

File: client/src/JobPaginatorState.js (hunk 2 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: had the same thought. this was in the old code. I wasn't sure why it was there, so I kept it there in case I broke something.

File: client/src/JobPaginatorState.js (hunk 6 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: Yes. I realized that after applying the unit tests. I reserved the logic

File: client/src/JobPaginatorZoom.jsx (hunk 5 above)
Comment (diboy2) [reply client/src/JobPaginatorZoom.jsx]: I think I should have kept it in the job paginator after reviewing the code.

File: client/src/useJobPaginator.js (hunk 8 above)
Comment (diboy2) [reply client/src/useJobPaginator.js]: I was trying to fix tsom things. I'm putting it back.

File: client/src/JobPaginator.jsx (hunk 1 above)
Comment (diboy2) [reply client/src/JobPaginator.jsx]: let me test it now.

File: client/src/JobPaginatorState.js (hunk 2 above)
Comment (demiankatz) [reply client/src/JobPaginatorState.js]: I believe that in the old code, the assignment was necessary because "category" was coming in as a function parameter and getting assigned to the state. In the new code, you're initializing the state via the "initialCategory" property, so there's no longer a need to reassign the value later.

File: client/src/JobPaginator.jsx [hunk 9]
```diff
@@ -1,326 +1,69 @@
...
+            <div className="row">
+                <JobPaginatorZoom zoom={zoom} getImageUrl={(size) => getImageUrl(order[currentPage], size)} />
+                <div className="six col">
+                    <p>
+                        {category} &gt; {job}
+                    </p>
+                    <PaginatorControls
```
Comment (diboy2) [inline client/src/JobPaginator.jsx:42]: This is still overly wordy for me. I'm planning to move some states and functions to a shared state, since they are redundantly used in the child components anyway.

File: client/src/JobPaginatorZoom.jsx (hunk 5 above)
Comment (diboy2) [reply client/src/JobPaginatorZoom.jsx]: actually, I'll move the column

File: client/src/JobPaginatorZoom.jsx (hunk 4 above)
Comment (diboy2) [reply client/src/JobPaginatorZoom.jsx]: Making change.

File: client/src/JobPaginatorZoom.jsx (hunk 5 above)
Comment (diboy2) [reply client/src/JobPaginatorZoom.jsx]: Made change.

File: client/src/JobPaginator.jsx (hunk 1 above)
Comment (diboy2) [reply client/src/JobPaginator.jsx]: made change. I'm not sure if it's okay or not.

File: client/src/JobPaginatorState.js (hunk 2 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: Okay. I removed it

File: client/src/JobPaginatorState.js (hunk 3 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: I made the change.

File: client/src/JobPaginator.jsx (hunk 9 above)
Comment (demiankatz) [reply client/src/JobPaginator.jsx:42]: Yes, that makes sense, though if you want to keep this as-is for now and work on that in a separate iteration, that would be fine with me. Might be good to get the current working state merged before making further revisions.

File: client/src/JobPaginator.jsx (hunk 1 above)
Comment (demiankatz) [reply client/src/JobPaginator.jsx]: There is a brief flash as the data loads, where the control switches from an empty state to a populated state... but I think that's okay for now. In the longer-term, I think a better solution is to make this control behave more correctly when the order is empty. There are some jobs which do not have any pages (e.g. audio jobs, PDF-only jobs), and it would be nice to have a more sensible display in those cases. I think it may make sense to pass the order.length into the PaginatorControls and hide all of the page-specific controls when there are no pages. Similarly, it may make sense for the PaginatorList to display a message like "No pages found." in place of the thumbnail list when pageCount is empty. This is not urgent, though -- maybe the best thing for now is to create a card for it and deal with it later.

File: client/src/JobPaginator.jsx [hunk 10]
```diff
@@ -1,326 +1,71 @@
...
+            deletePage,
+            save,
+            useEffect,
+            autonumberFollowingPages,
+            getImageUrl,
+            setZoom,
+        },
+    } = useJobPaginator(initialCategory, initialJob);
+
+    useEffect();
```
Comment (demiankatz) [inline client/src/JobPaginator.jsx]: What is the purpose of an empty call to useEffect()? I haven't seen that before (but I'm fairly new to this).

File: client/src/JobPaginator.test.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    });
+
+    it("renders", () => {
+        const wrapper = shallow(<JobPaginator {...props} />);
+        expect(wrapper.contains(props.initialCategory)).toBeTruthy();
+        expect(wrapper.contains(props.initialJob)).toBeTruthy();
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it("should loadJob from useEffect", async () => {
```
Comment (demiankatz) [inline client/src/JobPaginator.test.js:45]: Isn't this happening as a result of useJobPaginator rather than useEffect?

File: client/src/JobPaginatorState.js [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+const getStatus = (category, job) => {
+    const ajax = AjaxHelper.getInstance();
+    return new Promise((resolve) =>
+        ajax.getJSON(ajax.getJobUrl(category, job, "/status"), null, (data) => {
+            resolve(data);
+        })
+    );
+};
+
+const deleteImage = (imageUrl, success, error) => {
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js:32]: I just tested and found that the "delete image" feature doesn't work on this branch for some reason. I think that imageUrl is not getting constructed correctly, because when I look at my network console, I see DELETE requests going to URLs like: `http://localhost:9000/api/ingest/dime/job2/undefined/6`. Obviously, if you want to test this, you'll want to make a copy of your example job to test with so that you don't lose your original test data.

File: client/src/JobPaginatorState.js [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+        if (!getLabel(order, startIndex)) {
+            count++;
+        }
+    }
+    return count;
+};
+
+const validateSave = (order) => {
+    const count = countMagicLabels(order, 0);
+    return count > 0 && !confirmSavedMagicLabels(count);
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js:110]: The confirmSavedMagicLabels logic is inverted here; I have to hit "cancel" in order to save, and if I hit OK, the save action is canceled. (You can test this by trying to save a job with auto-generated labels).

Comment (demiankatz) [reply client/src/JobPaginatorState.js:110]: On a possibly-related note, apart from having inverted logic, the "Save" button does seem to work correctly. However, the "Save and Publish" button is more broken. I can't seem to find a way to publish a job at all, and during my attempts, I get this error:

<pre>
Warning: Cannot update a component (`JobPaginator`) while rendering a different component (`Thumbnail`). To locate the bad setState() call inside `Thumbnail`, follow the stack trace as described in https://fb.me/setstate-in-render
    in Thumbnail (at PaginatorList.jsx:18)
    in div (at PaginatorList.jsx:35)
    in PaginatorList (at JobPaginator.jsx:55)
    in div (at JobPaginator.jsx:36)
    in div (at JobPaginator.jsx:32)
    in div (at JobPaginator.jsx:31)
    in JobPaginator (at Routes.jsx:12)
    in JobPaginatorHook (at Routes.jsx:25)
    in Route (at Routes.jsx:24)
    in Switch (at Routes.jsx:17)
    in Routes (at VuDLPrep.jsx:15)
    in Router (created by BrowserRouter)
    in BrowserRouter (at VuDLPrep.jsx:14)
    in div (at VuDLPrep.jsx:10)
    in VuDLPrep (at src/index.js:14)
    in StrictMode (at src/index.js:13)
</pre>

This is followed by some weird, malformed API requests.

**Review by demiankatz — CHANGES_REQUESTED** [review]
@diboy2, I spent some more time testing and reviewing this PR today. It seems that the label-related functionality is now in very good shape, but some of the other features have issues. I haven't finished comprehensively reviewing all code, but since I found some significant bugs, I thought I would stop here and let you work on fixing these before I finish up, since I expect there will be some further changes now. Please let me know if you need help reproducing any of the issues I've reported!

File: client/src/JobPaginator.jsx (hunk 10 above)
Comment (diboy2) [reply client/src/JobPaginator.jsx]: useEffect replaces componentDidMount, and componentDidUpdate, I repurposed useEffect in useJobPaginator.

Comment (diboy2) [reply client/src/JobPaginator.jsx]: I think I'll just move loadJob into useEffect, to make this more explicit.

File: client/src/JobPaginatorState.js (hunk 12 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js:32]: Made change.

File: client/src/JobPaginator.jsx (hunk 10 above)
Comment (diboy2) [reply client/src/JobPaginator.jsx]: Moved it.

File: client/src/JobPaginatorState.js (hunk 13 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js:110]: I modified and made changes.

File: client/src/JobPaginator.jsx (hunk 1 above)
Comment (diboy2) [reply client/src/JobPaginator.jsx]: https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:54c0a65e7f5f491aa6f90021330eb74e@thread.tacv2_p_LH9rYXUCYke47Igm88R2rGQAGFab_h_1612357088580?tenantId=765a8de5-cf94-44f0-9caf-ae5bf8cfa366&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fboard%2Ftask%2F-fOapF7EMEyz0VaK771xXmQAI2KZ%22%2C%22channelId%22%3A%2219%3A54c0a65e7f5f491aa6f90021330eb74e%40thread.tacv2%22%7D

Comment (diboy2) [reply client/src/JobPaginator.jsx]: https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:54c0a65e7f5f491aa6f90021330eb74e@thread.tacv2_p_LH9rYXUCYke47Igm88R2rGQAGFab_h_1612357088580?tenantId=765a8de5-cf94-44f0-9caf-ae5bf8cfa366&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fboard%2Ftask%2F-fOapF7EMEyz0VaK771xXmQAI2KZ%22%2C%22channelId%22%3A%2219%3A54c0a65e7f5f491aa6f90021330eb74e%40thread.tacv2%22%7D

Comment (diboy2) [reply client/src/JobPaginator.jsx]: Created these two tasks.

File: client/src/JobPaginatorState.js [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+    return pages.reduce((order, page) => {
+        order.splice(findNewPagePosition(page, order), 0, {
+            filename: page,
+            label: null,
+        });
+        return order;
+    }, order);
+};
+
+const getNoneRemovedPages = (order, deletedPages) =>
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js]: Is this a typo in the function name? Should it be getNonRemovedPages? Or would getRemainingPages be more clear?

File: client/src/useJobPaginator.js [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+                },
+                () => {
+                    alert("Unable to delete!");
+                }
+            );
+        }
+    };
+
+    const save = async (published) => {
+        if (validateSave(order)) {
```
Comment (demiankatz) [inline client/src/useJobPaginator.js]: I think validateSave is another function in need of a better name. It's a bit confusing that it returns false when we should proceed and true when we should not -- I would expect the opposite from something with "validate" in the name. I would either reverse the return value of the function, or rename it to something more specific like `userAbortedSave` or `userMustReviewMoreLabels`.

File: client/src/JobPaginatorState.js [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+    deleteImage,
+    deletePageValidation,
+    getAddedPages,
+    getNoneRemovedPages,
+    getStatus,
+    getJob,
+    getLabel,
+    putJob,
+    confirmSavedMagicLabels,
+    findNewPagePosition,
```
Comment (demiankatz) [inline client/src/JobPaginatorState.js]: findNewPagePosition seems to be an internal support method; do we need to export it?

File: client/src/PaginatorControls.jsx [hunk 17]
```diff
@@ -32,13 +32,11 @@ class PaginatorControls extends React.Component {
             useMagic = true;
         }
         var label = this.labelInput.value;
-        return label.length === 0 && useMagic
-            ? this.props.paginator.getLabel(this.props.paginator.state.currentPage)
-            : label;
+        return label.length === 0 && useMagic ? this.props.getMagicLabel(this.props.currentPage) : label;
```
Comment (demiankatz) [inline client/src/PaginatorControls.jsx:35]: I see that this component is still class-based; I assume you plan to work on converting this in a future iteration.

File: client/src/Thumbnail.jsx [hunk 18]
```diff
@@ -1,22 +1,18 @@
...
     }, [selected]);
 
     return (
-        <div onClick={selectPage} className={"thumbnail" + (selected ? " selected" : "")} ref={ref}>
-            <img alt="" src={paginator.getImageUrl(number, "thumb")} />
+        <div onClick={() => setPage(number)} className={"thumbnail" + (selected ? " selected" : "")} ref={ref}>
+            <img alt="" src={getImageUrl(number, "thumb")} />
             <div className="number">{number + 1}</div>
-            <div className={"label" + (null === paginator.getLabel(number, false) ? " magic" : "")}>{label}</div>
+            <div className={"label" + (!getLabel(number) ? " magic" : "")}>{getMagicLabel(number)}</div>
```
Comment (demiankatz) [inline client/src/Thumbnail.jsx]: Is it safe to change the `null ===` condition to a simple `!` here? Is there a risk that falsy values will get misinterpreted as magic labels here and get the `magic` class inappropriately applied? (I tried to make this happen and couldn't, so it's probably fine -- but just wanted to raise the possibility in case there's some edge case you can think of). If this actually is a problem, there are a few other places that might need adjustment as well.

File: client/src/Thumbnail.test.js [hunk 19]
```diff
@@ -29,20 +28,21 @@ describe("Thumbnail", () => {
...
+        expect(props.getImageUrl).toHaveBeenCalledWith(props.number, "thumb");
         expect(wrapper.find(".thumbnail.selected").exists()).toBeTruthy();
         expect(wrapper.text().includes("13")).toBeTruthy();
         expect(wrapper.find(".label.magic").exists()).toBeTruthy();
     });
 
     it("renders correctly when not selected or no label", () => {
         props.selected = false;
-        props.paginator.getLabel.mockReturnValue("testLabel");
+        props.getLabel.mockReturnValue(true);
```
Comment (demiankatz) [inline client/src/Thumbnail.test.js]: It might be more realistic to also return "testLabel" here -- generally getLabel is either going to return null or the same value as getMagicLabel.

File: client/src/useJobPaginator.js [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+        if (deleted.length > 0) {
+            message.push(deleted.length + " file(s) have been removed from the job since the last edit.\n");
+            setOrder(getNoneRemovedPages(order, deleted));
+        }
+
+        if (added.length > 0) {
+            message.push(added.length + " file(s) have been added to the job since the last edit.\n");
+            setOrder(getAddedPages(order, added));
+        }
+        return message.length ? alert(message.join()) : undefined;
```
Comment (demiankatz) [inline client/src/useJobPaginator.js]: Does it really make sense to have a `return` here at all? While it's a tidy one-liner as written, it's also a bit unusual. I wonder if it would be more straightforward to just have a standard `if (message.length) { alert(message.join()); }` here.

File: client/src/useJobPaginator.test.js [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+            });
+
+            act(() => {
+                result.current.action.setLabel(0, "test2");
+            });
+
+            expect(result.current.state.order[0].label).toEqual("test2");
+        });
+
+        it("does not change the label", () => {
```
Comment (demiankatz) [inline client/src/useJobPaginator.test.js]: This is testing that set doesn't do anything if you specify an invalid index, correct? Maybe the description should incorporate that, like "does not change the label if index out of range".

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks again for all the work on this, @diboy2 -- I think it's very nearly done. Just a few very minor comments about naming, test descriptions, and possible (but I think unlikely) edge case issues.

File: client/src/JobPaginatorState.js (hunk 14 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: I thought nonRemovedPages was more appropriate?

File: client/src/JobPaginatorState.js (hunk 16 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: Oh, yes. I'll make that change.

File: client/src/PaginatorControls.jsx (hunk 17 above)
Comment (diboy2) [reply client/src/PaginatorControls.jsx:35]: Yes. JobPaginator is already large enough of a task. It's not in the scope.

File: client/src/JobPaginatorState.js (hunk 14 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: since deletedPages was one of the parameters

File: client/src/Thumbnail.jsx (hunk 18 above)
Comment (diboy2) [reply client/src/Thumbnail.jsx]: I'll change it back to null

File: client/src/Thumbnail.test.js (hunk 19 above)
Comment (diboy2) [reply client/src/Thumbnail.test.js]: this might have been from a previous test.

File: client/src/useJobPaginator.js (hunk 20 above)
Comment (diboy2) [reply client/src/useJobPaginator.js]: I don't know why I did it that way. I think it's because I watned it to return messages initially.

File: client/src/useJobPaginator.js (hunk 15 above)
Comment (diboy2) [reply client/src/useJobPaginator.js]: if(userMustReviewMoreLabels)

File: client/src/JobPaginatorState.js (hunk 14 above)
Comment (diboy2) [reply client/src/JobPaginatorState.js]: Changing spelling.

File: client/src/useJobPaginator.test.js (hunk 21 above)
Comment (diboy2) [reply client/src/useJobPaginator.test.js]: Made change.

**Review by demiankatz — APPROVED** [review]
